firefox/ScreenSharing/index.js

```diff
@@ -0,0 +1,35 @@
+var pageMod = require('sdk/page-mod');
+var prefsService = require('sdk/preferences/service');
+var allowDomainsPrefKey = 'media.getusermedia.screensharing.allowed_domains';
+var gDomains = ['*.example.com'];
+
+exports.main = function (options) {
+
+  if (options.loadReason === 'install' || 'enable') {
+    var curPrefs = prefsService.get(allowDomainsPrefKey).replace(/\s/g, '').split(',');
+
+    gDomains.forEach(function(domain){
+      if (curPrefs.indexOf(domain) !== -1) {
+        return;
+      }
+      curPrefs.push(domain);
+    });
+    prefsService.set(allowDomainsPrefKey, curPrefs.join(','));
+  }
+};
+
+exports.onUnload = function (reason) {
+  if (reason === 'uninstall' || 'disable') {
```

Comment:
As above. This type of expression will actually always be true. I think this highlights the need to do a bit of qa on this before it goes out.